test/lib/ufe/testRename.py

```diff
@@ -420,14 +420,10 @@ def testRenameNotifications(self):
         ufeObs = TestObserver()
 
         # We start off with no observers
-        self.assertFalse(ufe.Scene.hasObjectAddObserver(ufeObs))
-        self.assertFalse(ufe.Scene.hasObjectDeleteObserver(ufeObs))
-        self.assertFalse(ufe.Scene.hasObjectPathChangeObserver(ufeObs))
-
-        # Add the UFE observers we want to test
-        ufe.Scene.addObjectAddObserver(ufeObs)
-        ufe.Scene.addObjectDeleteObserver(ufeObs)
-        ufe.Scene.addObjectPathChangeObserver(ufeObs)
+        self.assertFalse(ufe.Scene.hasObserver(ufeObs))
+
+        # Add the UFE observer we want to test
+        ufe.Scene.addObserver(ufeObs)
```

Comment:
@ppt-adsk  

Adopting to new UFE interface changes for testRename. Simply calling Subject's "addObserver" and "hasObserver" once.

Reply:
I'm a bit confused here. When you call addObserver() now what types of notifs are you observing for? Previously we would only get add/delete/pathchange notifs. Now what will we get?

Reply:
@seando-adsk 

The only notification used in this test case is always ObjectRename. We know that because that's what the test is doing ( testRenameNotifications ). See Line 35

Other notifications in this case are irrelevant. I don't think we even need to any checks for other notification types ( e.g ObjectAdd, ObjectDelete) in TestObserver ( 

Previously we were adding 3 observers to the scene (ObjectAdd, ObjectDelet, ObjectPathChange) when we only just needed one.

Reply:
Actually no. If you look at the original TestObserver you see that it was making sure that during the rename we only received an UFE ObjectRename, and that we did **not** receive any ObjectAdd or ObjectDelete. 

In this test we did not ask for SubtreeInvalidate observation. So if this test somehow triggered a UFE subtree invalidate notif this observer would not get that notif since it never observed on it. Has that behavior changed now and when you observe on the scene you will get all of the scene notifs?

Reply:
In the original test, you are adding 3 observers but when are the other scene notifications(e.g ObjectAdd, ObjectDelete) are sent?

Scene still gets all of the "scene notifs" as I am demonstrating in **testObservableScene.py**

The Scene notifications are ObjectAdd, ObjectDelete, ObjectPathChange, SubtreeInvalidate and the newly added Composition 

In the new design, Scene adds the observer **only once** since it drives directly from subject. Scene can also now `publicly calls notify`.

Sorry, if this is still confusing, we can talk internally and I can walk you through the though process.